tests/dragonfly/replication_test.py

```diff
@@ -1838,31 +1838,62 @@ async def test_replicaof_reject_on_load(df_local_factory, df_seeder_factory):
     replica.stop()
 
 
+# note: please be careful if you want to change any of the parameters used in this test.
+# changing parameters without extensive testing may easily lead to weak testing case assertion
+# which means eviction may not get triggered.
 @pytest.mark.asyncio
 @pytest.mark.slow
 async def test_eviction_propagation(df_local_factory, df_seeder_factory):
-    master = df_local_factory.create(proactor_threads=1, cache_mode="true", maxmemory="290mb")
+    master = df_local_factory.create(
+        proactor_threads=1, cache_mode="true", maxmemory="267mb", enable_heartbeat_eviction="false"
+    )
     replica = df_local_factory.create(proactor_threads=1)
     df_local_factory.start_all([master, replica])
 
     c_master = master.client()
     c_replica = replica.client()
 
-    seeder = df_seeder_factory.create(port=master.port, val_size=1000, stop_on_failure=False)
-
     await c_master.execute_command("DEBUG POPULATE 6200000")
     await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
     await wait_available_async(c_replica, timeout=3000)
 
-    # Check data after full sync
+    # Fill more data in stable state
+    seeder = df_seeder_factory.create(port=master.port, val_size=1000, stop_on_failure=False)
+    await seeder.run(target_ops=50000)
```

Comment:
instead of defining target ops which could lead maybe to failures when changing a littel bit the numbers of maxmemory and debug populate keys number I would do 
```
seeder.run()
while True:
sleep(0.3)
info = await c_master.info("stats")
if info["evicted_keys"] > 0
  break
seeder.stop()

```
you can also add a timeout to the while and assert we dont get to this timeout

Reply:
this requires a asyncio.create, you can't run like that with seeder.run() seeder.stop(), see the latest code.